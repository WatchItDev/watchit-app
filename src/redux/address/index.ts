import { createSlice, PayloadAction } from '@reduxjs/toolkit';

interface AddressState {
  address: string;
  profileId: string;
  showRainbow: boolean;
}

const initialState: AddressState = {
  address: '',
  profileId: '',
  showRainbow: false,
};

const addressSlice = createSlice({
  name: 'address',
  initialState,
  reducers: {
    storeAddress(state, action: PayloadAction<{ address: string; profileId: string }>) {
      state.address = action.payload.address;
      state.profileId = action.payload.profileId;
    },

    toggleRainbow(state) {
      state.showRainbow = !state.showRainbow;
    },

  },
});

export const { storeAddress, toggleRainbow } = addressSlice.actions;
export default addressSlice.reducer;
