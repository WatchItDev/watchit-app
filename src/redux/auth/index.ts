import {createSlice, PayloadAction} from "@reduxjs/toolkit";

export interface AuthReducerState {
  session: any;
  isSessionLoading: boolean;
  isLoginModalOpen: boolean;
  balance: number;
  currentStep: number;
  isUpdatingMetadata: boolean;
  email: string;
}

const initialState: AuthReducerState = {
  session: null,
  isSessionLoading: false,
  isLoginModalOpen: false,
  balance: 0,
  currentStep: 0,
  isUpdatingMetadata: false,
  email: "",
};

const authSlice = createSlice({
  name: "auth",
  initialState,
  reducers: {
    openLoginModal: (state) => {
      state.isLoginModalOpen = true;
    },
    closeLoginModal: (state) => {
      state.isLoginModalOpen = false;
    },
    setAuthLoading: (state, action: PayloadAction<Pick<AuthReducerState, "isSessionLoading">>) => {
      state.isSessionLoading = action.payload.isSessionLoading;
    },
    setBalance: (state, action: PayloadAction<Pick<AuthReducerState, "balance">>) => {
      state.balance = action.payload.balance;
    },
    setProfileCreationStep: (state, action: PayloadAction<{step: number}>) => {
      state.currentStep = action.payload.step;
    },
    resetCurrentStep: (state) => {
      state.currentStep = 0;
    },
    setSession: (state, action: PayloadAction<{session: any}>) => {
      state.session = action.payload.session;
    },
    setEmail: (state, action: PayloadAction<string>) => {
      state.email = action.payload;
    },
    updateProfileData: (
      state,
      action: PayloadAction<{
        name: string;
        bio: string;
        profileImage?: string;
        backgroundImage?: string;
      }>,
    ) => {
      if (state.session && state.session.profile) {
        state.session.profile.metadata = {
          ...state.session.profile.metadata,
          displayName: action.payload.name,
          bio: action.payload.bio,
          picture: {
            optimized: {
              uri: action.payload.profileImage ? action.payload.profileImage : undefined,
            },
          },
          coverPicture: {
            optimized: {
              uri: action.payload.backgroundImage ? action.payload.backgroundImage : undefined,
            },
          },
        };
      }
    },
    setIsUpdatingMetadata: (state, action: PayloadAction<boolean>) => {
      state.isUpdatingMetadata = action.payload;
    },
  },
});

export const {
  openLoginModal,
  closeLoginModal,
  setAuthLoading,
  setBalance,
  setProfileCreationStep,
  resetCurrentStep,
  setSession,
  updateProfileData,
  setIsUpdatingMetadata,
  setEmail,
} = authSlice.actions;

export default authSlice.reducer;
