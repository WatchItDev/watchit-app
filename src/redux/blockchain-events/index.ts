import { createSlice, PayloadAction } from '@reduxjs/toolkit'
import { Log } from 'viem'

interface BlockchainEventsState {
  events: Log[];
}

const initialState: BlockchainEventsState = {
  events: [],
}

const blockchainEventsSlice = createSlice({
  name: 'blockchainEvents',
  initialState,
  reducers: {
    addBlockchainEvent(state, action: PayloadAction<Log>) {
      state.events.push(action.payload)
    },
    setBlockchainEvents(state, action: PayloadAction<Log[]>) {
      state.events = action.payload
    },
    clearBlockchainEvents(state) {
      state.events = []
    },
  },
})

export const { addBlockchainEvent, setBlockchainEvents, clearBlockchainEvents } =
  blockchainEventsSlice.actions

export default blockchainEventsSlice.reducer
