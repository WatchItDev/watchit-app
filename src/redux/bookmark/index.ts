import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { Post } from '@src/graphql/generated/graphql.ts';

export interface BookmarkReducerState {
  bookmarkPublications: Post[];
  hiddenBookmarks: Post[];
}

const initialState: BookmarkReducerState = {
  bookmarkPublications: [],
  hiddenBookmarks: [],
};

const bookmarkSlice = createSlice({
  name: 'bookmark',
  initialState,
  reducers: {
    // Add a post to bookmarks and remove it from hidden if it is
    addBookmark: (state, action: PayloadAction<Post>) => {
      const isAlreadyBookmarked = state.bookmarkPublications.some(
        (publication) => publication.id === action.payload.id
      );

      if (!isAlreadyBookmarked) {
        state.bookmarkPublications.push(action.payload);
        // Remove from hiddenBookmarks if exists
        state.hiddenBookmarks = state.hiddenBookmarks.filter(
          (publication) => publication.id !== action.payload.id
        );
      }
    },

    // Move a post to hiddenBookmarks instead of deleting it completely
    removeBookmark: (state, action: PayloadAction<string>) => {
      const publicationToHide = state.bookmarkPublications.find(
        (publication) => publication.id === action.payload
      );

      if (publicationToHide) {
        state.hiddenBookmarks.push(publicationToHide);
        state.bookmarkPublications = state.bookmarkPublications.filter(
          (publication) => publication.id !== action.payload
        );
      }
    },

    // Remove all bookmarks
    clearBookmarks: (state) => {
      state.bookmarkPublications = [];
    },
  },
});

export const { addBookmark, removeBookmark, clearBookmarks } = bookmarkSlice.actions;

export default bookmarkSlice.reducer;
