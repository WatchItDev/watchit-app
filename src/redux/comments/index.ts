import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { AnyPublication } from '@lens-protocol/api-bindings';

type PendingComment = AnyPublication & { uri: string };

export type CommentsReducerState = {
  refetchTriggerByPublication: {
    [publicationId: string]: number;
  };
  hiddenComments: AnyPublication[];
  counterLikes: { [publicationId: string]: number };
  comments: { [publicationId: string]: AnyPublication[] };
  pendingComments: { [publicationId: string]: PendingComment[] };
};

const initialState: CommentsReducerState = {
  refetchTriggerByPublication: {},
  hiddenComments: [],
  counterLikes: {},
  comments: {},
  pendingComments: {},
};

const commentsSlice = createSlice({
  name: 'comments',
  initialState,
  reducers: {
    refetchCommentsByPublication: (state, action: PayloadAction<string>) => {
      const publicationId = action.payload;
      if (!state.refetchTriggerByPublication[publicationId]) {
        state.refetchTriggerByPublication[publicationId] = 1;
      } else {
        state.refetchTriggerByPublication[publicationId] += 1;
      }
    },
    hiddeComment: (state, action: PayloadAction<AnyPublication>) => {
      state.hiddenComments.push(action.payload);
    },
    setCounterLikes: (state, action: PayloadAction<{ publicationId: string; likes: number }>) => {
      state.counterLikes[action.payload.publicationId] = action.payload.likes;
    },
    incrementCounterLikes: (state, action: PayloadAction<string>) => {
      const publicationId = action.payload;
      if (state.counterLikes[publicationId] !== undefined) {
        state.counterLikes[publicationId] += 1;
      }
    },
    decrementCounterLikes: (state, action: PayloadAction<string>) => {
      const publicationId = action.payload;
      if (state.counterLikes[publicationId] !== undefined) {
        state.counterLikes[publicationId] -= 1;
      }
    },
    addPendingComment: (
      state,
      action: PayloadAction<{ publicationId: string; comment: PendingComment }>
    ) => {
      const { publicationId, comment } = action.payload;
      if (!state.pendingComments[publicationId]) {
        state.pendingComments[publicationId] = [];
      }
      // Prepend new comment to the beginning of the list
      state.pendingComments[publicationId] = [comment, ...state.pendingComments[publicationId]];
    },
    removePendingComment: (
      state,
      action: PayloadAction<{ publicationId: string; commentId: string }>
    ) => {
      const { publicationId, commentId } = action.payload;

      // Delete the comment from the pending list
      state.pendingComments[publicationId] = state.pendingComments[publicationId].filter(
        (comment) => comment.id !== commentId
      );
    },
  },
});

export const {
  hiddeComment,
  refetchCommentsByPublication,
  setCounterLikes,
  incrementCounterLikes,
  decrementCounterLikes,
  addPendingComment,
  removePendingComment,
} = commentsSlice.actions;

export default commentsSlice.reducer;
