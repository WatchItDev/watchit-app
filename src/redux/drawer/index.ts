import { createSlice } from '@reduxjs/toolkit'

const initialState = {
  open: false,
}

const drawerSlice = createSlice({
  name: 'drawer',
  initialState,
  reducers: {
    openDrawer: (state) => {
      state.open = true
    },
    closeDrawer: (state) => {
      state.open = false
    },
    toggleDrawer: (state) => {
      state.open = !state.open
    },
  },
})

export const { openDrawer, closeDrawer, toggleDrawer } = drawerSlice.actions

export default drawerSlice.reducer
