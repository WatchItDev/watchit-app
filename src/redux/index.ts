export { minibarInitialState } from "@redux/minibar";
export { drawerInitialState } from "@redux/drawer";
export { authInitialState } from "@redux/auth";
export { bookmarkInitialState } from "@redux/bookmark";
export { commentsInitialState } from "@redux/comments";
export { notificationsInitialState } from "@redux/notifications";
export { followersInitialState } from "@redux/followers";
export { addressInitialState } from "@redux/address";
export { blockchainEventsInitialState } from "@redux/blockchain-events";
export { transactionInitialState } from "@redux/transactions";
