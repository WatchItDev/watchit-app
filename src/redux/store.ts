import { type Store } from 'redux';
import { configureStore } from '@reduxjs/toolkit';
import rootReducer from '@src/redux/reducer';

export const createStore = (initialState = {}): Store => {
  return configureStore({
    reducer: rootReducer,
    devTools: true,
    preloadedState: initialState,
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware({
        serializableCheck: false,
      }),
  });
};

export type RootState = ReturnType<typeof store.getState>;

export const store = createStore();
