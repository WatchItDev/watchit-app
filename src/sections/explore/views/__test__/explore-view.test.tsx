import { describe, expect, it } from "vitest";
import { render } from "@testing-library/react";
import ExploreView from "../explore-view";
import { Provider } from "react-redux";
import { store } from "@src/redux/store";
import { MemoryRouter } from "react-router";
import { MockedProvider } from "@apollo/client/testing";
import { exploreViewMock } from "./__mock__/exploreView.mock";

vi.mock("@src/workers/backgroundTaskWorker?worker", () => {
  return {
    default: class {
      postMessage() {}
      terminate() {}
      addEventListener() {}
      removeEventListener() {}
    },
  };
});

const renderWithProviders = () => {
  return render(
    <Provider store={store}>
      <MockedProvider mocks={exploreViewMock} addTypename={false}>
        <MemoryRouter future={{ v7_startTransition: true, v7_relativeSplatPath: true }}>
          <ExploreView />
        </MemoryRouter>
      </MockedProvider>
    </Provider>,
  );
};

describe("Testing in the <ExploreView/> component", () => {
  it("should match snapshot", () => {
    const { container } = renderWithProviders();
    expect(container).toMatchSnapshot();
  });

  // it("should render all public sections of ExploreView", () => {
  //   vi.spyOn(authHook, "useAuth").mockReturnValue({ isFullyAuthenticated: true });
  //   renderWithProviders();
  //   expect(screen.getByText(/Latest creators/i)).toBeInTheDocument();
  //   expect(screen.getByText(/Publications/i)).toBeInTheDocument();
  // });
  //
  // it("should not render bookmarks when user is not authenticated", () => {
  //   vi.spyOn(authHook, "useAuth").mockReturnValue({ isFullyAuthenticated: false });
  //   renderWithProviders();
  //   expect(screen.queryByText(/Bookmarks/i)).not.toBeInTheDocument();
  // });
  //
  // it("should render bookmarks when user is authenticated", () => {
  //   vi.spyOn(authHook, "useAuth").mockReturnValue({ isFullyAuthenticated: true });
  //   renderWithProviders();
  //   expect(screen.getByText(/Bookmarks/i)).toBeInTheDocument();
  // });
});
