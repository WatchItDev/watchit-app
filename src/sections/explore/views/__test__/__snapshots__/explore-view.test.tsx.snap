// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing in the <ExploreView/> component > should match snapshot 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1hb8xwj-MuiContainer-root"
  >
    <div
      class="MuiStack-root css-1waufmq-MuiStack-root"
    >
      <div
        class="MuiBox-root css-1r5plea"
      >
        <div
          class="MuiBox-root css-13nz920"
        >
          <div
            class="MuiBox-root css-tk1iyj"
          >
            <span
              class="MuiLinearProgress-root MuiLinearProgress-colorInherit MuiLinearProgress-indeterminate css-yptbrg-MuiLinearProgress-root"
              role="progressbar"
            >
              <span
                class="MuiLinearProgress-bar MuiLinearProgress-barColorInherit MuiLinearProgress-bar1Indeterminate css-jlm0hp-MuiLinearProgress-bar1"
              />
              <span
                class="MuiLinearProgress-bar MuiLinearProgress-barColorInherit MuiLinearProgress-bar2Indeterminate css-1wsh2fw-MuiLinearProgress-bar2"
              />
            </span>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-14btkih"
      >
        <div
          class="MuiStack-root css-nen11g-MuiStack-root"
          data-testid="carouselArrowsContainer"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-oxvw91-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              data-icon="eva:arrow-ios-forward-fill"
              data-sx="{"transform":" scaleX(-1)"}"
              data-testid="leftButton"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="slick-slider slick-initialized"
          >
            <div
              class="slick-list"
              style="height: 0px;"
            >
              <div
                class="slick-track"
                style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
              />
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-17oewfz-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              data-icon="eva:arrow-ios-forward-fill"
              data-sx="{}"
              data-testid="rightButton"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardHeader-root css-1pkklwm-MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
            >
              Publications
            </span>
          </div>
          <div
            class="MuiCardHeader-action css-sgoict-MuiCardHeader-action"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              data-testid="leftButton"
              tabindex="0"
              type="button"
            >
              <svg
                class="tabler-icon tabler-icon-chevron-left "
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15 6l-6 6l6 6"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              data-testid="rightButton"
              tabindex="0"
              type="button"
            >
              <svg
                class="tabler-icon tabler-icon-chevron-right "
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 6l6 6l-6 6"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiCardContent-root css-fd8ij7-MuiCardContent-root"
        >
          <div
            class="MuiBox-root css-1v61vtn"
          >
            <div
              class="slick-slider slick-initialized"
            >
              <div
                class="slick-list"
                style="height: 0px;"
              >
                <div
                  class="slick-track"
                  style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
